vm based providers: Bump CNAO to v0.100.0-rc0

oshoval opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:
1. Consider creating this PR as draft: https://github.com/kubevirt/kubevirt/blob/main/CONTRIBUTING.md#consider-opening-your-pull-request-as-draft
2. Follow the instructions for writing a release note from k8s: https://git.k8s.io/community/contributors/guide/release-notes.md
-->

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
`hack/bump-cnao.sh` is broken, worth to fix it on follow-up.

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:
<!--  Write your release note:
1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
2. If no release note is required, just write "NONE".
-->
```release-note
None
```

orelmisan commented on cluster-provision/gocli/opts/cnao/manifests/operator.yaml:
```diff
@@ -489,11 +517,12 @@ spec:
         - --secure-listen-address=:8443
         - --upstream=http://127.0.0.1:8080
-        image: quay.io/openshift/origin-kube-rbac-proxy@sha256:e2def4213ec0657e72eb790ae8a115511d5b8f164a62d3568d2f1bff189917e8
+        image: quay.io/brancz/kube-rbac-proxy@sha256:e6a323504999b2a4d2a6bf94f8580a050378eba0900fd31335cf9df5787d9a9b
```
Could you please elaborate why has this registry changed? (seems to be someones personal registry).
Same above and bellow.

oshoval replied:
info here https://github.com/kubevirt/cluster-network-addons-operator/pull/1917
it is safe, this is the original repo https://github.com/openshift/kube-rbac-proxy is forked from

orelmisan reviewed (commented):
Thank you for the PR @oshoval.
Just one little question.

oshoval commented:
1.31 passed
1.32 hanged because of problem not related to this PR - we can just retest until it runs
1.33 - still running

orelmisan reviewed (approved):
Thank you @oshoval.

I'm in favor of this change as it will enable us to use Multus v4 across all k/k network lanes. 
While our larger goal remains deploying network components without relying on an intermediate component like CNAO, the current challenges with manually updating Multus make this an important interim step.

oshoval commented:
Thanks

@dhiller @enp0s3 can you please take a look ?

enp0s3 commented:
/approve

oshoval commented:
addressed 
```
quay.io/cephcsi/cephcsi:v3.14.2
+ echo '(Please add them to file /home/prow/go/src/github.com/kubevirt/kubevirtci/cluster-provision/k8s/1.31/extra-pre-pull-images)'
(Please add them to file /home/prow/go/src/github.com/kubevirt/kubevirtci/cluster-provision/k8s/1.31/extra-pre-pull-images)
```
and rebased

oshoval commented:
/retest

enp0s3 commented:
Hi, I see that the job was stable on another PRs. Can you please try to reproduce it manually? it doesn't look to me as a flake.
